### PR TITLE
Cbp 4082 - Adding target argument in kaniko action

### PR DESCRIPTION
**Pull request by Akashsami4456**
Added target argument in kaniko to build particular stage in multi stage docker file.

### CODE REVIEW

File: .cloudbees/testing/action.yml [hunk 1]
```diff
@@ -71,12 +75,13 @@ runs:
   steps:
     - id: imgbuild
       name: Build and publish Docker image
-      uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/actions/kaniko-action:${{ action.scm.sha }}
+      # uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/actions/kaniko-action:${{ action.scm.sha }}
+      uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/actions/kaniko-action:897eb954bffc9d26fe4b7a4fbe1085335ed4142c
```
Comment (SrimanPadmanabanCB) [inline .cloudbees/testing/action.yml:79]: Revert this line, Keep the existing line

File: action.yml [hunk 2]
```diff
@@ -71,12 +75,13 @@ runs:
   steps:
     - id: imgbuild
       name: Build and publish Docker image
-      uses: docker://public.ecr.aws/l7o7z1g8/actions/kaniko-action:${{ action.scm.sha }}
+      # uses: docker://public.ecr.aws/l7o7z1g8/actions/kaniko-action:${{ action.scm.sha }}
+      uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/actions/kaniko-action:897eb954bffc9d26fe4b7a4fbe1085335ed4142c
```
Comment (SrimanPadmanabanCB) [inline action.yml:79]: Revert this line, Keep the existing line

File: internal/kaniko/execute.go [hunk 3]
```diff
@@ -224,6 +224,11 @@ func (k *Config) cmdBuilder(digestFile string) (*exec.Cmd, error) {
 		cmdArgs = append(cmdArgs, "--skip-default-registry-fallback")
 	}
 
+	if k.Target != "" {
+		fmt.Printf("Test:%v", k.Target)
```
Comment (SrimanPadmanabanCB) [inline internal/kaniko/execute.go:228]: Dont add test in logs, Print the proper message

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (SrimanPadmanabanCB) [inline test.sh:1]: Do you need this file for testing ? If not remove this.

File: .cloudbees/testing/action.yml (hunk 1 above)
Comment (Akashsami4456) [reply .cloudbees/testing/action.yml:79]: I reverted this line in below commit.

File: internal/kaniko/execute.go (hunk 3 above)
Comment (Akashsami4456) [reply internal/kaniko/execute.go:228]: Included proper message instead of Test.

File: test.sh (hunk 4 above)
Comment (Akashsami4456) [reply test.sh:1]: Removed this test file

File: Dockerfile-Test [hunk 5]
```diff
@@ -0,0 +1,24 @@
+FROM node:18 AS dev
```
[the remark this answers is not in the document]
Comment (Akashsami4456) [reply Dockerfile-Test:1]: Removed this multi stage test docker file in below commit.

**Review by SrimanPadmanabanCB — APPROVED** [review]

**Review by stephenc — APPROVED** [review]

File: internal/kaniko/types.go [hunk 6]
```diff
@@ -18,6 +18,8 @@ type Config struct {
 	SkipDefaultRegistryFallback bool `json:"skipDefaultRegistryFallback,omitempty"`
 	// Verbosity is the verbosity level of the Kaniko executor.
 	Verbosity string `json:"verbosity,omitempty"`
+	// Target field allows you to build a particular stag in multistage docker files.
```
Comment (pinaki124) [inline internal/kaniko/types.go:21]: misspelt `stage`.

**Pull request merged**